### PR TITLE
Merge sync-baseTemplates branch

### PR DESCRIPTION
**Pull request by vidurb**
Contains the addition of the base, init and html layout templates and Cache Flag plugin.

### CODE REVIEW

File: templates/_layouts/html.twig [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+{% block html %}
+  <!DOCTYPE html>
+  <html lang="{{ craft.app.language }}" class="has-no-js">
+  <head>
+    {%- spaceless %}
+      <meta charset="utf-8"/>
+
+      {% for sheet in styles %}
+    <link rel="stylesheet" href="{{ sheet }}">
```
Comment (rungta) [inline templates/_layouts/html.twig]: Fix formatting

File: templates/_layouts/html.twig [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  {% endblock %}
+
+
+
+  {% for script in scripts %}
+    <script async src="{{ script }}"></script>
+  {% endfor %}
+
+  {% block body_end %}{% endblock %}
+  </body>
```
Comment (rungta) [inline templates/_layouts/html.twig:49]: Fix formatting (contents of `<body>` should be indented).

File: templates/_layouts/init.twig [hunk 3]
```diff
@@ -0,0 +1,85 @@
+{% spaceless %}
+
+  {#
+    Initialize Global Env
+  #}
+
+  {% set cacheableEnv = (not craft.app.request.isLivePreview) and craft.app.request.getParam('token') == NULL %}
+  {% set cacheableEnv = cacheableEnv and craft.app.request.getParam('nocache') is same as(NULL) %}
+  {% set cacheableEnv = cacheableEnv and craft.app.request.isPost()|length == 0 %}
```
Comment (rungta) [inline templates/_layouts/init.twig]:
```suggestion
  {% set cacheableEnv = cacheableEnv and not craft.app.request.isPost %}
```

**Review by rungta — CHANGES_REQUESTED** [review]

File: templates/_layouts/html.twig (hunk 1 above)
Comment (vidurb) [reply templates/_layouts/html.twig]: Fixed in next-to-next commit.

File: templates/_layouts/html.twig (hunk 2 above)
Comment (vidurb) [reply templates/_layouts/html.twig:49]: Fixed in next-to-next commit.

File: templates/_layouts/html.twig (hunk 1 above)
Comment (rungta) [reply templates/_layouts/html.twig]: Have you pushed that upstream? I’m not seeing the fix.

Comment (vidurb) [reply templates/_layouts/html.twig]: Should be there now, if it wasn't before.

File: templates/_layouts/init.twig [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    base_asset_url ~ 'build/style.purged.min.' ~ assetManifest['style.purged.min.css'] ~ '.css',
+  ] %}
+  {% set scripts = [
+    base_asset_url ~ 'build/urgent.min.' ~ assetManifest['urgent.min.js'] ~ '.js',
+    base_asset_url ~ 'build/deferred.min.' ~ assetManifest['deferred.min.js'] ~ '.js',
+  ] %}
+
+
+
+
```
Comment (rungta) [inline templates/_layouts/init.twig]:
```suggestion
```

**Review by rungta — APPROVED** [review]

**Pull request merged**